src/ostree/ot-builtin-commit.c

```diff
@@ -97,104 +97,64 @@ static GOptionEntry options[] = {
 };
 
 static gboolean
-parse_statoverride_file (GHashTable   **out_mode_add,
-                         GCancellable  *cancellable,
-                         GError        **error)
+parse_file_by_line (const char    *path,
+                    gboolean     (*cb)(const char*, void*, GError**),
+                    void          *cbdata,
+                    GCancellable  *cancellable,
+                    GError       **error)
 {
   gboolean ret = FALSE;
-  gsize len;
-  char **iter = NULL; /* nofree */
-  g_autoptr(GHashTable) ret_hash = NULL;
-  g_autoptr(GFile) path = NULL;
   g_autofree char *contents = NULL;
+  g_autoptr(GFile) file = NULL;
   char **lines = NULL;
 
-  path = g_file_new_for_path (opt_statoverride_file);
-
-  if (!g_file_load_contents (path, cancellable, &contents, &len, NULL,
-                             error))
+  file = g_file_new_for_path (path);
+  if (!g_file_load_contents (file, cancellable, &contents, NULL, NULL, error))
     goto out;
-  
-  ret_hash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
-  lines = g_strsplit (contents, "\n", -1);
-
-  for (iter = lines; iter && *iter; iter++)
-    {
-      const char *line = *iter;
-
-      if (*line == '+')
-        {
-          const char *spc;
-          guint mode_add;
 
-          spc = strchr (line + 1, ' ');
-          if (!spc)
-            {
-              g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
-                           "Malformed statoverride file");
-              goto out;
-            }
-          
-          mode_add = (guint32)(gint32)g_ascii_strtod (line + 1, NULL);
-          g_hash_table_insert (ret_hash,
-                               g_strdup (spc + 1),
-                               GUINT_TO_POINTER((gint32)mode_add));
-        }
-    }
+  lines = g_strsplit (contents, "\n", -1);
+  for (char **iter = lines; iter && *iter; iter++)
```

Comment:
I'd be a bit more comfortable if we had braces `{` for nested loops/conditionals...otherwise it's too easy down the line to add an additional statement and have it go in the wrong section.